src/main/webapp/assets/js/index.js

```diff
@@ -14,13 +14,68 @@ async function getMultipleMockData() {
 async function initHomePage() {
   const mockData = await getMultipleMockData();
   const productsToLoad = Math.min(INITIAL_PRODUCT_LOAD, mockData.length);
-  let foodLocations = []
+
+  getFilterCategories(mockData)
+
+  let foodLocations = [];
 
   for (let i = 0; i < productsToLoad; i++) {
     document.getElementById("product-listings").innerHTML += homePageListingTemplate(mockData[i]);
-    foodLocations.push(createLocation(mockData[i]))
+    foodLocations.push(createLocation(mockData[i]));
+  }
+  initMap(foodLocations);
+}
+
+/**
+ * Obtain a list of filter categories to allow product listings to be filtered accordingly.
+ */
+async function getFilterCategories(mockData) {
+  var categories = []
+  mockData.forEach(data => {
+    categories = [...categories, ...data.categories]
+  });
+
+  var categories = new Set(categories)
+  document.getElementById("filter-buttons").innerHTML += addCategoryFilters('All');
```

Comment:
consider using appendChild instead of innerHTML += (the latter is considered by some to be bad code)

https://stackoverflow.com/questions/11515383/why-is-element-innerhtml-bad-code

Reply:
i'll fix this in my PR! @jacoblipech

Reply:
Oh I just realised it's not in my branch since this isn't merged yet. I'll fix it up for homepageListingTemplate only!